pkg/resources/apiserver/service.go

```diff
@@ -87,7 +87,10 @@ func ServiceReconciler(exposeStrategy kubermaticv1.ExposeStrategy, externalURL s
 			se.Spec.Ports[0].Port = 443
 			if exposeStrategy == kubermaticv1.ExposeStrategyTunneling {
 				se.Spec.Ports[0].TargetPort = intstr.FromInt(resources.APIServerSecurePort)
-				se.Spec.Ports[0].NodePort = 0 // allows switching from other expose strategies
+				// set nodeport as zero only if the service type is ClusterIP.
```

Comment:
And now we can see that the comment is not really adding anything at all and can be removed alltogether :) And I promise it's my last nitpick 😁